addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/handler/KNXBasicThingHandler.java

```diff
@@ -0,0 +1,682 @@
+                    }
+
+                    // check if there is a Device Object in the KNX Actor
+                    Thread.sleep(OPERATION_INTERVAL);
+                    byte[] elements = getBridgeHandler().readDeviceProperties(address, DEVICE_OBJECT, PID.OBJECT_TYPE,
+                            0, 1, false, OPERATION_TIMEOUT);
+                    if ((elements == null ? 0 : toUnsigned(elements)) == 1) {
+
+                        Thread.sleep(OPERATION_INTERVAL);
+                        String ManufacturerID = Manufacturer
+                                .getName(toUnsigned(getBridgeHandler().readDeviceProperties(address, DEVICE_OBJECT,
+                                        PID.MANUFACTURER_ID, 1, 1, false, OPERATION_TIMEOUT)));
+                        Thread.sleep(OPERATION_INTERVAL);
+                        String serialNo = DataUnitBuilder.toHex(getBridgeHandler().readDeviceProperties(address,
+                                DEVICE_OBJECT, PID.SERIAL_NUMBER, 1, 1, false, OPERATION_TIMEOUT), "");
+                        Thread.sleep(OPERATION_INTERVAL);
+                        String hardwareType = DataUnitBuilder.toHex(getBridgeHandler().readDeviceProperties(address,
+                                DEVICE_OBJECT, HARDWARE_TYPE, 1, 1, false, OPERATION_TIMEOUT), " ");
+                        Thread.sleep(OPERATION_INTERVAL);
+                        String firmwareRevision = Integer
+                                .toString(toUnsigned(getBridgeHandler().readDeviceProperties(address, DEVICE_OBJECT,
+                                        PID.FIRMWARE_REVISION, 1, 1, false, OPERATION_TIMEOUT)));
+
+                        Map<String, String> properties = editProperties();
+                        properties.put(MANUFACTURER_NAME, ManufacturerID);
+                        properties.put(MANUFACTURER_SERIAL_NO, serialNo);
+                        properties.put(MANUFACTURER_HARDWARE_TYPE, hardwareType);
+                        properties.put(MANUFACTURER_FIRMWARE_REVISION, firmwareRevision);
+                        try {
+                            updateProperties(properties);
+                        } catch (Exception e) {
+                            // TODO : ignore for now, but for Things created through the DSL, this should also NOT throw
+                            // an exception! See forum discussions
+                        }
+                        logger.info("Identified device {} as a {}, type {}, revision {}, serial number {}", address,
+                                ManufacturerID, hardwareType, firmwareRevision, serialNo);
+                    } else {
+                        logger.warn("The KNX Actor with address {} does not expose a Device Object", address);
+                    }
+
+                    // TODO : According to the KNX specs, devices should expose the PID.IO_LIST property in the DEVICE
```

Comment:
There is a TODO, what should be done and can it be completed before merging the PR?